src/Mjolnir.ts

```diff
@@ -863,14 +895,20 @@ export class Mjolnir {
 
             // Iterate all the enabled protections
             for (const protection of this.enabledProtections) {
+                let consequence: Consequence | undefined = undefined;
```

Comment:
This flow assumes that we're only ever going to have a single consequence.

Maybe we should rather make it an EventEmitter event, with Mjolnir (or in the future some `ProtectionManager`) listening to e.g. `protection.consequence` and according accordingly?

Reply:
Definitely.